### PR TITLE
refactor tests

### PR DESCRIPTION
**Pull request by mforets**
this is just to split the tests folder into a file for the problems and a file for the actual tests.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
       matrix:
         julia-version: ['1.5']
         julia-arch: [x64]
-        os: [ubuntu-latest, macOS-latest, windows-latest]
+        os: [ubuntu-latest, windows-latest]
```
Comment (mforets) [inline .github/workflows/ci.yml:24]: i think we're good testing in ubuntu and windows..

File: test/problems.jl [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# ---------------------------------------------------------
```
Comment (mforets) [inline test/problems.jl:1]: the functions in this file `problems.jl` now return the tuple (initial value problem, analytic solution)

File: test/problems.jl [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    lelem = 1.0 / nelem
+    Area  = .25
+    C = rho * csh * lelem * Area * ( [0.5+0.5+0.0 0; 0  0.0+0.5+0.5] )
+    K = kco *       lelem * Area * ( [1.0+1.0 -1.; -1. 1.0+1.0] * 1.0 / lelem^2 )
+    M = zeros(2, 2)
+    R = zeros(2)
+    sys = SecondOrderAffineContinuousSystem(M, C, K, R)
+
+    # build initial state
+    U₀ = zeros(2)
```
Comment (mforets) [inline test/problems.jl:43]: @jorgepz does it make sense to write `zeros(nelem - 1)`  here?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jorgepz) [reply .github/workflows/ci.yml:24]: so do i

File: test/problems.jl (hunk 3 above)
Comment (jorgepz) [reply test/problems.jl:43]: short answer: it does.
long answer: since the problem is 1D the number of nodes is nelem+1, and since dirichlet boundary conditions are applied in both ends, the size of the vector of unknowns is nelem+1-2.

**Review by jorgepz — APPROVED** [review]
